doc/types.md

```diff
@@ -377,3 +380,34 @@ in the model or other values also.
 ```
 
 This will show the "secretSchema" when the value at `this.model.owner` equals "secret".
+
+The schema function also allows the returning of Promises. e.g.
+
+```js
+
+  let model = {
+    name: 'Tom',
+    language: 'es'
+  };
+
+  let userSchema = [{
+    key: 'name',
+  }, {
+    key   : 'language',
+    type  : 'computed',
+    schema: () => {
+      return Promise((resolve, reject) => {
+        http.get('languages').then(languages => {
```

Comment:
in this example you can just return the  http.get('languages').then(l... directly. no need to another promise or catching